hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -438,9 +438,8 @@ private void processAppendResult(AppendResult result, List<HoodieRecord> recordL
           .getColumnsToIndex(hoodieTable.getMetaClient().getTableConfig(),
               config.getMetadataConfig(), Lazy.eagerly(Option.of(writeSchemaWithMetaFields)),
               Option.of(this.recordMerger.getRecordType())));
-      final List<Schema.Field> fieldsToIndex = writeSchemaWithMetaFields.getFields().stream()
-          .filter(field -> columnsToIndexSet.contains(field.name()))
-          .collect(Collectors.toList());
+      final List<Pair<String, Schema.Field>> fieldsToIndex = columnsToIndexSet.stream()
```

Comment:
So, the fieldname in the pair looks like `a.b.c` for nested field and the value i.e. schema is only for child `c` or is it the whole schema for `a.b.c`?

Reply:
yes, key will be "a.b.c" and schema.field will refer to the datatype of "c" since thats what we might need to know. a: what col name to use while encoding data to col stats partition. and the schema field for leaf field ("c" in this context).